src/css/hiddenVisibleSelectors.js

```diff
@@ -4,12 +4,10 @@ define([
 ], function( jQuery ) {
 
 jQuery.expr.filters.hidden = function( elem ) {
-	// Use OR instead of AND as the element is not visible if either is true
-	// See tickets #10406 and #13132
-	return !elem.offsetWidth || !elem.offsetHeight;
+	return !jQuery.expr.filters.visible( elem );
```

Comment:
The browser probably does that anyway, and even if it does take another function call the DOM queries will be much more cruel.